libsql-server/src/replication/primary/logger.rs

```diff
@@ -242,14 +250,22 @@ impl LogFile {
 
     pub fn push_page(&mut self, page: &WalPage) -> anyhow::Result<()> {
         let checksum = self.compute_checksum(page);
+        let data = if let Some(encryption) = &self.encryption {
+            let mut data = BytesMut::with_capacity(page.data.len());
```

Comment:
we can probably reuse some buffer here since we have an exclusive ref to self

Reply:
@MarinPostma do you mean the buffer in page.data? I could, that means taking WalPage by a mut ref and making it keep BytesMut, to encrypt in place. Or is there another buffer you have in mind?

Reply:
no I was thinking about adding a buffer in self. but actually maybe we can have a mutable page

Reply:
or just take ownership `WalPage`... let me try that

Reply:
ok, added a single encryption buffer; my original idea of encrypting in-place would be more intrusive (classic Arc vs contagious-lifetimes-everywhere issue), so I'd rather skip it for now